R/lazyframe__lazy.R

```diff
@@ -265,15 +265,71 @@ LazyFrame_filter = "use_extendr_wrapper"
 
 #' @title New DataFrame from LazyFrame_object$collect()
 #' @description collect DataFrame by lazy query
+#' @param type_coercion Boolean. Do type coercion optimization.
+#' @param predicate_pushdown  Boolean. Do predicate pushdown optimization.
+#' @param projection_pushdown  Boolean. Do projection pushdown optimization.
+#' @param simplify_expression  Boolean. Run simplify expressions optimization.
+#' @param no_optimization  Boolean. Turn off (certain) optimizations.
+#' @param slice_pushdown  Boolean. Slice pushdown optimization.
+#' @param common_subplan_elimination  Boolean. Will try to cache branching subplans that occur on
+#' self-joins or unions.
+#' @param streaming  Boolean. Run parts of the query in a streaming fashion
+#' (this is in an alpha state)
+#' @param collect_in_background Boolean. Detach this query from R session. Computation will start
+#' in background. Get a handle which later can be converted into the resulting DataFrame. Useful
+#' in interactive mode to not lock R session.
+#' @details
+#' Note: use `$fetch()` if you want to run your query on the first `n` rows only.
+#' This can be a huge time saver in debugging queries.
 #' @keywords LazyFrame DataFrame_new
-#' @return collected `DataFrame`
+#' @return collected `DataFrame` or if colkect
 #' @examples pl$DataFrame(iris)$lazy()$filter(pl$col("Species") == "setosa")$collect()
-LazyFrame_collect = function() {
-  unwrap(.pr$LazyFrame$collect_handled(self), "in $collect():")
+LazyFrame_collect = function(
+  type_coercion = TRUE,
+  predicate_pushdown = TRUE,
+  projection_pushdown = TRUE,
+  simplify_expression = TRUE,
+  no_optimization = FALSE,
+  slice_pushdown = TRUE,
+  common_subplan_elimination = TRUE,
+  streaming = FALSE,
+  collect_in_background = FALSE
+) {
+
+  if (isTRUE(no_optimization)) {
+    predicate_pushdown = FALSE
+    projection_pushdown = FALSE
+    slice_pushdown = FALSE
+    common_subplan_elimination = FALSE
+  }
+
+  if (isTRUE(streaming)) {
+    common_subplan_elimination = FALSE
+  }
+
+  collect_f = if( isTRUE(collect_in_background)) {
+    .pr$LazyFrame$collect_background
+  } else {
+    .pr$LazyFrame$collect
+  }
+
+  self |>
+    .pr$LazyFrame$optimization_toggle(
+      type_coercion,
+      predicate_pushdown,
+      projection_pushdown,
+      simplify_expression,
+      slice_pushdown,
+      common_subplan_elimination,
+      streaming
+    ) |>
+    and_then(collect_f) |>
+    unwrap("in $collect():")
 }
 
 #' @title New DataFrame from LazyFrame_object$collect()
-#' @description collect DataFrame by lazy query
+#' @description collect DataFrame by lazy query (SOFT DEPRECATED)
+#' @details This function is soft deprecated. Use $collect(collect_in_background = TRUE) instead
```

Comment:
I don't understand this doc, how is this different from `$collect()`? I know it gets deprecated but until it's removed it will appear in the docs so it might be good to clarify. Also the example uses `collect()` instead of `collect_in_background()`

Reply:
This documentation and behavior will be updated alot in #311. I suggest to just leave it as is here.